### PR TITLE
Allow to override project properties from command line

### PR DESCRIPTION
**Pull request by ssheikin**

### CODE REVIEW

**Comment by ssheikin** [comment]
@jvanzyl @ppalucha @nineinchnick @wendigo please review

File: provisio-maven-plugin/src/main/java/ca/vanzyl/maven/plugins/provisio/Provisio.java [hunk 1]
```diff
@@ -95,8 +98,13 @@ public List<Runtime> findDescriptorsForPackagingTypeInExtensionRealms(MavenProje
 
   public Runtime parseDescriptor(InputStream inputStream, MavenProject project) {
```
Comment (wendigo) [inline provisio-maven-plugin/src/main/java/ca/vanzyl/maven/plugins/provisio/Provisio.java:99]: Maybe: 
```
public static Properties getPropertiesWithSystemOverrides(MavenProject project) {
    Properties properties = new Properties(project.getProperties());
    properties.putAll(System.getProperties());
    return properties;
  }

  public Runtime parseDescriptor(InputStream inputStream, MavenProject project) {
    RuntimeReader parser = new RuntimeReader(Actions.defaultActionDescriptors(), versionMap(project));
    Map<String, String> variables = Maps.newHashMap();
    variables.putAll((Map) getPropertiesWithSystemOverrides(project));
    variables.put("project.version", project.getVersion());
    variables.put("project.groupId", project.getGroupId());
    variables.put("project.artifactId", project.getArtifactId());
    variables.put("project.build.directory", project.getBuild().getDirectory());
    variables.put("project.basedir", project.getBasedir().getAbsolutePath());
    variables.put("basedir", project.getBasedir().getAbsolutePath());
    return parser.read(inputStream, variables);
  }
```

?

Comment (ssheikin) [reply provisio-maven-plugin/src/main/java/ca/vanzyl/maven/plugins/provisio/Provisio.java:99]: I didn't want to put all properties to the context, but let it be.

**Comment by ssheikin** [comment]
@jvanzyl could you please take a look?

**Comment by jvanzyl** [comment]
I added you all so that you can review and merge changes.

**Comment by jvanzyl** [comment]
I'll cut a release on Monday.

**Comment by jvanzyl** [comment]
Yes, I can cut a release. I'll also notify Sonatype to add you to the project so that you can do releases as well.

**Comment by kokosing** [comment]
Any update with the release?

**Comment by jvanzyl** [comment]
I'm not in a place with a computer where I can do a release. I'm happy to give you access at oss.sonatype.org so you can make release. I just need to submit a JIRA ticket giving you access. You wan to cut the release?

**Comment by kokosing** [comment]
Yes. Please :)

**Comment by ssheikin** [comment]
Sure! I'd be happy to cut the release.

**Comment by jvanzyl** [comment]
@kokosing has access now to release. I don't know who you are personally (at least I don't think) so you'll have to wait for myself or @kokosing

**Comment by ssheikin** [comment]
Thank you, that's perfectly fine. I didn't have that info on time of writing.

**Pull request merged**